### PR TITLE
[CfgEditor] Check properties existence in loop

### PR DESCRIPTION
**Pull request by dayo09**
This commit checks properties' existence in for loop.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

---

For https://github.com/Samsung/ONE-vscode/issues/1624#issuecomment-1720750787

### CODE REVIEW

File: media/CfgEditor/index.js [hunk 1]
```diff
@@ -347,35 +347,39 @@ function registerOptimizeOptions() {
...
-    cellName.setAttribute("grid-column", "2");
-    row.appendChild(cellName);
-
-    let cellDescription = document.createElement("vscode-data-grid-cell");
-    cellDescription.textContent = oneOptimizationList[optName].description;
-    cellDescription.setAttribute("grid-column", "3");
-    row.appendChild(cellDescription);
-
-    basicOptimizeTable.appendChild(row);
+    if (Object.prototype.hasOwnProperty.call(oneOptimizationList, optName)) {
```
Comment (shs-park) [inline media/CfgEditor/index.js:350]: I don't know which one you prefer, but I suggest that you can reduce the indentation and improve readability like below.

```suggestion
if (!Object.prototype.hasOwnProperty.call(oneOptimizationList, optName)) {
  continue;
}
```

**Review by shs-park — DISMISSED** [review]
LGTM!
=)

**Review by glistening — APPROVED** [review]
LGTM

File: media/CfgEditor/index.js (hunk 1 above)
Comment (dayo09) [reply media/CfgEditor/index.js]: I agree :-D Thanks for the suggestion 😄

File: media/CfgEditor/index.js [hunk 2]
```diff
@@ -347,35 +347,41 @@ function registerOptimizeOptions() {
...
-
-    let cellDescription = document.createElement("vscode-data-grid-cell");
-    cellDescription.textContent = oneOptimizationList[optName].description;
-    cellDescription.setAttribute("grid-column", "3");
-    row.appendChild(cellDescription);
-
-    basicOptimizeTable.appendChild(row);
+if (!Object.prototype.hasOwnProperty.call(oneOptimizationList, optName)) {
+  continue;
+}
```
Comment (dayo09) [inline media/CfgEditor/index.js:352]:
```suggestion
    if (!Object.prototype.hasOwnProperty.call(oneOptimizationList, optName)) {
      continue;
    }
```

File: media/CfgEditor/index.js (hunk 1 above)
Comment (shs-park) [reply media/CfgEditor/index.js]: Oh, I missed to clean up the braces and code indentation. sorry 🥲

**Review by shs-park — APPROVED** [review]

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**